### PR TITLE
Cleanup CI for documentation

### PR DESCRIPTION
**Pull request by glrf**
## Summary

This PR cleans up the CI for documentation changes.

* Tests (including e2e tests) and code linter are not run in CI if the PR only changes documentation
* Code Linting does not generate documentation anymore
* Remove Make targets:
  * `make docs-<html/pdf/...>`: Was broken since `v1.2.0` and not used as final documentation webpage is generated separately
  * `make docs-syntax`: Was broken since `v1.2.0`
  * `make docs-htmltest`: Was broken since `v1.2.0`
  * Meta tags to generate downloadable docs
* Removed any dependency other than podman/docker for working with documentation
  * Auto detects if using docker or podman
  * Auto detects if `go` is present on machine. Otherwise uses `golang` container for auto-generating documentation
* Add documentation linting step to CI that runs vale (only if documentation changes)

See https://github.com/k8up-io/k8up/pull/870 for a PR that only touches documentation (and fixes vale linting issues)

### For Code changes

- [x] Categorize the PR by setting a good title and adding one of the labels:
  `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
  as they show up in the changelog
- [x] I have not made _any_ changes in the `charts/` directory.

### CODE REVIEW

File: docs/docs.mk [hunk 1]
```diff
@@ -1,99 +1,51 @@
...
 .PHONY: docs-preview
 docs-preview: ## Start documentation preview at http://localhost:2020 with Live Reload
 	$(preview_cmd)
 
-.PHONY: docs-html-open
-docs-html-open: docs-html ## Start documentation preview at http://localhost:2020 with Live Reload and open in browser
-	$(shell xdg-open _public/index.html)
-
-.PHONY: docs-html docs-pdf docs-manpage docs-epub docs-kindle
-docs-html:    $(out_dir)/index.html ## Generate HTML version of documentation with Antora, output at ./_public/
```
Comment (zugao) [inline docs/docs.mk:80]: There targets are not needed anymore?

Comment (glrf) [reply docs/docs.mk:80]: As explained in the summary. No, and they were broken for 2 years without bothering anyone

**Review by zugao — APPROVED** [review]
LGTM

**Comment by glrf** [comment]
E2E test is flaking again, but as we did not change anything that can effect it, I'm merging.

**Pull request merged**